File: Assignments/Assignment_53/program1.cpp
// Write generic program to copy one array into another.

#include<iostream>
using namespace std;

template<class T>

void CopyArray(T *src, T *dest, int iSize)
{
    int iCnt = 0;

    for(iCnt = 0; iCnt < 5; iCnt++)
    {
        dest[iCnt] = src[iCnt];
    }
}

int main()
{
    int arr[5] = {10,20,30,40,50};

    int CopyArr[5];
    int iCnt = 0;

    CopyArray(arr,CopyArr,5);
    
    cout<<"Copied Array is : ";
    
    for(iCnt = 0; iCnt < 5; iCnt++)
    {
        cout<<CopyArr[iCnt]<<" ";
    }
    cout<<"\n";

    return 0;
}

File: Assignments/Assignment_57/program1.java
//  Write java program to accept two files names from user and open firsr file and create new file (second name) and copy the data from first file into newly created file.

import java.util.Scanner;
import java.io.*;

class program1
{
    public static void main(String A[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter Source file name : ");
        String src = sc.nextLine();

        System.out.print("Enter destination file name : ");
        String dest = sc.nextLine();

        FileInputStream fin = new FileInputStream(src);
        FileOutputStream fout = new FileOutputStream(dest);

        int ch;

        while((ch = fin.read()) != -1)
        {
            fout.write(ch);
        }

        fin.close();
        fout.close();

        System.out.println("File copied successfully");
        sc.close();
    }
}